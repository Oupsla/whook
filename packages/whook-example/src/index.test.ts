import { constant } from 'knifecycle';
import {
  runServer,
  prepareServer,
  prepareEnvironment as basePrepareEnvironment,
} from './index';
import axios from 'axios';
import YError from 'yerror';

const packageConf = require('../../../package');

describe('runServer', () => {
  const logger = {
    info: jest.fn(),
    error: jest.fn(),
  };
  const debug = jest.fn();
  const time = jest.fn();
  const exit = jest.fn();
  const PORT = 9999;
  const HOST = 'localhost';
  const BASE_PATH = '/v4';

  async function prepareEnvironment() {
    const $ = await basePrepareEnvironment();

    $.register(constant('API_VERSION', packageConf.version));
    $.register(constant('BASE_PATH', BASE_PATH));
    $.register(
      constant('BASE_ENV', {
        DEV_MODE: '1',
      }),
    );
    $.register(constant('PORT', PORT));
    $.register(constant('HOST', HOST));
    $.register(constant('NODE_ENV', 'test'));
    $.register(constant('DEBUG_NODE_ENVS', []));
    $.register(constant('NODE_ENVS', ['test']));
    $.register(constant('exit', exit));
    $.register(constant('time', time));
    $.register(constant('logger', logger));
    $.register(constant('debug', debug));

    return $;
  }
  process.env.ISOLATED_ENV = '1';

  let $instance;

  beforeAll(async () => {
    const { $instance: _instance } = await runServer(
      prepareEnvironment,
      prepareServer,
      ['$instance', 'httpServer', 'process'],
    );

    $instance = _instance;
  }, 5000);

  afterAll(async () => {
    await $instance.destroy();
  }, 1000);

  afterEach(() => {
    time.mockReset();
    debug.mockReset();
    logger.info.mockReset();
    logger.error.mockReset();
  });

  it('should work', async () => {
    expect({
      debugCalls: debug.mock.calls.map(filterPaths),
      logInfoCalls: logger.info.mock.calls.map(filterPaths),
      logErrorCalls: logger.error.mock.calls.map(filterPaths),
    }).toMatchSnapshot();
  });

  it('should ping', async () => {
    time.mockReturnValue(new Date('2014-01-26T00:00:00.000Z').getTime());

    const { status, headers, data } = await axios({
      method: 'get',
      url: `http://${HOST}:${PORT}${BASE_PATH}/ping`,
    });

    expect({
      status,
      headers: {
        ...headers,
        // Erasing the Date header that may be added by Axios :/
        date: undefined,
      },
      data,
      debugCalls: debug.mock.calls.map(filterPaths),
      logInfoCalls: logger.info.mock.calls.map(filterPaths),
      logErrorCalls: logger.error.mock.calls.map(filterPaths),
    }).toMatchSnapshot();
  });

  it('should authenticate users', async () => {
    time.mockReturnValue(new Date('2014-01-26T00:00:00.000Z').getTime());

    const { status, headers, data } = await axios({
      method: 'get',
      url: `http://${HOST}:${PORT}${BASE_PATH}/diag`,
      headers: {
        authorization: `Fake 1-admin`,
      },
    });

    expect({
      status,
      headers: {
        ...headers,
        // Erasing the Date header that may be added by Axios :/
        date: undefined,
      },
      data,
      debugCalls: debug.mock.calls.map(filterPaths),
      logInfoCalls: logger.info.mock.calls.map(filterPaths),
      logErrorCalls: logger.error.mock.calls.map(filterPaths),
    }).toMatchSnapshot();
  });

  it('should fail with bad fake tokens', async () => {
    time.mockReturnValue(new Date('2014-01-26T00:00:00.000Z').getTime());

    try {
      await axios({
        method: 'get',
        url: `http://${HOST}:${PORT}${BASE_PATH}/diag`,
        headers: {
          authorization: `Fake e-admin`,
        },
      });
      throw new YError('E_UNEXPECTED_SUCCESS');
    } catch (err) {
      const { status, headers, data } = err.response;

      expect({
        status,
        headers: {
          ...headers,
          // Erasing the Date header that may be added by Axios :/
          date: undefined,
        },
        data,
        debugCalls: debug.mock.calls.map(filterPaths),
        logInfoCalls: logger.info.mock.calls.map(filterPaths),
        logErrorCalls: logger.error.mock.calls
          .map(filterPaths)
          .filter(([arg1]) => arg1 !== 'An error occured'),
      }).toMatchSnapshot();
    }
  });
});

function filterPaths(strs) {
  return strs.map(str =>
    'string' !== typeof str
      ? str
      : str.replace(
          /('| )(\/[^/]+){1,}\/whook\//g,
          '$1/home/whoiam/projects/whook/',
        ),
  );
}
