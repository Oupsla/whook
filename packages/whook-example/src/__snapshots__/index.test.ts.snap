// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runServer should authenticate users 1`] = `
Object {
  "data": Object {
    "transactions": Object {
      "1": Object {
        "errored": false,
        "id": "1",
        "ip": "127.0.0.1",
        "method": "GET",
        "protocol": "http",
        "reqHeaders": Object {
          "accept": "application/json, text/plain, */*",
          "authorization": "Fake 1-admin",
          "connection": "close",
          "host": "localhost:9999",
          "user-agent": "axios/0.19.0",
        },
        "startInBytes": 164,
        "startOutBytes": 0,
        "startTime": 1390694400000,
        "url": "/v4/diag",
      },
    },
  },
  "debugCalls": Array [
    Array [
      "‚è≥ - Created a delay:",
      30000,
    ],
    Array [
      "‚è≥ - Cleared a delay",
    ],
  ],
  "headers": Object {
    "access-control-allow-headers": "Accept,Accept-Encoding,Accept-Language,Referrer,Content-Type,Content-Encoding,Authorization,Keep-Alive,User-Agent",
    "access-control-allow-methods": "GET,POST,PUT,DELETE,OPTIONS",
    "access-control-allow-origin": "*",
    "connection": "close",
    "content-type": "application/json",
    "date": undefined,
    "transaction-id": "1",
    "transfer-encoding": "chunked",
    "vary": "Origin",
    "x-authenticated": "{\\"hash\\":\\"1-admin\\",\\"userId\\":1,\\"scopes\\":[\\"admin\\"]}",
  },
  "logErrorCalls": Array [],
  "logInfoCalls": Array [
    Array [
      Object {
        "endInBytes": 164,
        "endOutBytes": 826,
        "endTime": 1390694400000,
        "errored": false,
        "id": "1",
        "ip": "127.0.0.1",
        "method": "GET",
        "protocol": "http",
        "reqHeaders": Object {
          "accept": "application/json, text/plain, */*",
          "authorization": "Fake 1-admin",
          "connection": "close",
          "host": "localhost:9999",
          "user-agent": "axios/0.19.0",
        },
        "resHeaders": Object {
          "Access-Control-Allow-Headers": "Accept,Accept-Encoding,Accept-Language,Referrer,Content-Type,Content-Encoding,Authorization,Keep-Alive,User-Agent",
          "Access-Control-Allow-Methods": "GET,POST,PUT,DELETE,OPTIONS",
          "Access-Control-Allow-Origin": "*",
          "Vary": "Origin",
          "X-Authenticated": "{\\"hash\\":\\"1-admin\\",\\"userId\\":1,\\"scopes\\":[\\"admin\\"]}",
          "content-type": "application/json",
        },
        "startInBytes": 164,
        "startOutBytes": 0,
        "startTime": 1390694400000,
        "statusCode": 200,
        "url": "/v4/diag",
      },
    ],
  ],
  "status": 200,
}
`;

exports[`runServer should fail with bad fake tokens 1`] = `
Object {
  "data": Object {
    "error": Object {
      "code": "E_INVALID_FAKE_TOKEN",
      "guruMeditation": "2",
    },
  },
  "debugCalls": Array [
    Array [
      "‚è≥ - Created a delay:",
      30000,
    ],
    Array [
      "‚è≥ - Cleared a delay",
    ],
  ],
  "headers": Object {
    "access-control-allow-headers": "Accept,Accept-Encoding,Accept-Language,Referrer,Content-Type,Content-Encoding,Authorization,Keep-Alive,User-Agent",
    "access-control-allow-methods": "GET,POST,PUT,DELETE,OPTIONS",
    "access-control-allow-origin": "*",
    "cache-control": "private",
    "connection": "close",
    "content-type": "application/json",
    "date": undefined,
    "transaction-id": "2",
    "transfer-encoding": "chunked",
    "vary": "Origin",
  },
  "logErrorCalls": Array [],
  "logInfoCalls": Array [
    Array [
      Object {
        "endInBytes": 164,
        "endOutBytes": 510,
        "endTime": 1390694400000,
        "errored": true,
        "id": "2",
        "ip": "127.0.0.1",
        "method": "GET",
        "protocol": "http",
        "reqHeaders": Object {
          "accept": "application/json, text/plain, */*",
          "authorization": "Fake e-admin",
          "connection": "close",
          "host": "localhost:9999",
          "user-agent": "axios/0.19.0",
        },
        "resHeaders": Object {
          "Access-Control-Allow-Headers": "Accept,Accept-Encoding,Accept-Language,Referrer,Content-Type,Content-Encoding,Authorization,Keep-Alive,User-Agent",
          "Access-Control-Allow-Methods": "GET,POST,PUT,DELETE,OPTIONS",
          "Access-Control-Allow-Origin": "*",
          "Vary": "Origin",
          "cache-control": "private",
          "content-type": "application/json",
        },
        "startInBytes": 164,
        "startOutBytes": 0,
        "startTime": 1390694400000,
        "statusCode": 400,
        "url": "/v4/diag",
      },
    ],
  ],
  "status": 400,
}
`;

exports[`runServer should ping 1`] = `
Object {
  "data": Object {
    "pong": "pong",
  },
  "debugCalls": Array [
    Array [
      "‚è≥ - Created a delay:",
      30000,
    ],
    Array [
      "üîì - Public endpoint detected, letting the call pass through!",
    ],
    Array [
      "‚è≥ - Cleared a delay",
    ],
  ],
  "headers": Object {
    "access-control-allow-headers": "Accept,Accept-Encoding,Accept-Language,Referrer,Content-Type,Content-Encoding,Authorization,Keep-Alive,User-Agent",
    "access-control-allow-methods": "GET,POST,PUT,DELETE,OPTIONS",
    "access-control-allow-origin": "*",
    "connection": "close",
    "content-type": "application/json",
    "date": undefined,
    "transaction-id": "0",
    "transfer-encoding": "chunked",
    "vary": "Origin",
    "x-node-env": "test",
  },
  "logErrorCalls": Array [],
  "logInfoCalls": Array [
    Array [
      Object {
        "endInBytes": 135,
        "endOutBytes": 447,
        "endTime": 1390694400000,
        "errored": false,
        "id": "0",
        "ip": "127.0.0.1",
        "method": "GET",
        "protocol": "http",
        "reqHeaders": Object {
          "accept": "application/json, text/plain, */*",
          "connection": "close",
          "host": "localhost:9999",
          "user-agent": "axios/0.19.0",
        },
        "resHeaders": Object {
          "Access-Control-Allow-Headers": "Accept,Accept-Encoding,Accept-Language,Referrer,Content-Type,Content-Encoding,Authorization,Keep-Alive,User-Agent",
          "Access-Control-Allow-Methods": "GET,POST,PUT,DELETE,OPTIONS",
          "Access-Control-Allow-Origin": "*",
          "Vary": "Origin",
          "X-Node-ENV": "test",
          "content-type": "application/json",
        },
        "startInBytes": 135,
        "startOutBytes": 0,
        "startTime": 1390694400000,
        "statusCode": 200,
        "url": "/v4/ping",
      },
    ],
  ],
  "status": 200,
}
`;

exports[`runServer should work 1`] = `
Object {
  "debugCalls": Array [
    Array [
      "üë£ - Logging service initialized.",
    ],
    Array [
      "‚åõ - Delay service initialized.",
    ],
    Array [
      "‚ôªÔ∏è - Loading the environment service.",
    ],
    Array [
      "‚û∞ - Plugin \\"@whook/cli\\" resolved to: /home/whoiam/projects/whook/packages/whook-cli/dist",
    ],
    Array [
      "‚û∞ - Plugin \\"@whook/whook\\" resolved to: /home/whoiam/projects/whook/dist",
    ],
    Array [
      "üè≠ - Initializing the CONFIGS service.",
    ],
    Array [
      "ü§ñ - Initializing the \`$autoload\` service.",
    ],
    Array [
      "üö´ - Service \\"MAX_HEADERS_COUNT\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/MAX_HEADERS_COUNT",
    ],
    Array [
      "üö´ - Service \\"MAX_HEADERS_COUNT\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/MAX_HEADERS_COUNT",
    ],
    Array [
      "üö´ - Service \\"MAX_HEADERS_COUNT\\" not found in: /home/whoiam/projects/whook/dist/services/MAX_HEADERS_COUNT",
    ],
    Array [
      "üö´ - Service \\"KEEP_ALIVE_TIMEOUT\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/KEEP_ALIVE_TIMEOUT",
    ],
    Array [
      "üö´ - Service \\"KEEP_ALIVE_TIMEOUT\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/KEEP_ALIVE_TIMEOUT",
    ],
    Array [
      "üö´ - Service \\"KEEP_ALIVE_TIMEOUT\\" not found in: /home/whoiam/projects/whook/dist/services/KEEP_ALIVE_TIMEOUT",
    ],
    Array [
      "üö´ - Service \\"TIMEOUT\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/TIMEOUT",
    ],
    Array [
      "üö´ - Service \\"TIMEOUT\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/TIMEOUT",
    ],
    Array [
      "üö´ - Service \\"TIMEOUT\\" not found in: /home/whoiam/projects/whook/dist/services/TIMEOUT",
    ],
    Array [
      "üö´ - Service \\"MAX_CONNECTIONS\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/MAX_CONNECTIONS",
    ],
    Array [
      "üö´ - Service \\"MAX_CONNECTIONS\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/MAX_CONNECTIONS",
    ],
    Array [
      "üö´ - Service \\"MAX_CONNECTIONS\\" not found in: /home/whoiam/projects/whook/dist/services/MAX_CONNECTIONS",
    ],
    Array [
      "üö´ - Service \\"PROCESS_NAME\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/PROCESS_NAME",
    ],
    Array [
      "üö´ - Service \\"PROCESS_NAME\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/PROCESS_NAME",
    ],
    Array [
      "üö´ - Service \\"PROCESS_NAME\\" not found in: /home/whoiam/projects/whook/dist/services/PROCESS_NAME",
    ],
    Array [
      "üö´ - Service \\"SIGNALS\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/SIGNALS",
    ],
    Array [
      "üö´ - Service \\"SIGNALS\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/SIGNALS",
    ],
    Array [
      "üö´ - Service \\"SIGNALS\\" not found in: /home/whoiam/projects/whook/dist/services/SIGNALS",
    ],
    Array [
      "üö´ - Service \\"BUFFER_LIMIT\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/BUFFER_LIMIT",
    ],
    Array [
      "üö´ - Service \\"BUFFER_LIMIT\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/BUFFER_LIMIT",
    ],
    Array [
      "üö´ - Service \\"BUFFER_LIMIT\\" not found in: /home/whoiam/projects/whook/dist/services/BUFFER_LIMIT",
    ],
    Array [
      "üíø - Service \\"API\\" found in: /home/whoiam/projects/whook/packages/whook-example/src/services/API.ts",
    ],
    Array [
      "üö´ - Service \\"PARSERS\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/PARSERS",
    ],
    Array [
      "üö´ - Service \\"PARSERS\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/PARSERS",
    ],
    Array [
      "üö´ - Service \\"PARSERS\\" not found in: /home/whoiam/projects/whook/dist/services/PARSERS",
    ],
    Array [
      "üö´ - Service \\"STRINGIFYERS\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/STRINGIFYERS",
    ],
    Array [
      "üö´ - Service \\"STRINGIFYERS\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/STRINGIFYERS",
    ],
    Array [
      "üö´ - Service \\"STRINGIFYERS\\" not found in: /home/whoiam/projects/whook/dist/services/STRINGIFYERS",
    ],
    Array [
      "üö´ - Service \\"DECODERS\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/DECODERS",
    ],
    Array [
      "üö´ - Service \\"DECODERS\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/DECODERS",
    ],
    Array [
      "üö´ - Service \\"DECODERS\\" not found in: /home/whoiam/projects/whook/dist/services/DECODERS",
    ],
    Array [
      "üö´ - Service \\"ENCODERS\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/ENCODERS",
    ],
    Array [
      "üö´ - Service \\"ENCODERS\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/ENCODERS",
    ],
    Array [
      "üö´ - Service \\"ENCODERS\\" not found in: /home/whoiam/projects/whook/dist/services/ENCODERS",
    ],
    Array [
      "üö´ - Service \\"QUERY_PARSER\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/QUERY_PARSER",
    ],
    Array [
      "üö´ - Service \\"QUERY_PARSER\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/QUERY_PARSER",
    ],
    Array [
      "üö´ - Service \\"QUERY_PARSER\\" not found in: /home/whoiam/projects/whook/dist/services/QUERY_PARSER",
    ],
    Array [
      "üö´ - Service \\"DEV_ACCESS_TOKEN\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/DEV_ACCESS_TOKEN",
    ],
    Array [
      "üö´ - Service \\"DEV_ACCESS_TOKEN\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/DEV_ACCESS_TOKEN",
    ],
    Array [
      "üö´ - Service \\"DEV_ACCESS_TOKEN\\" not found in: /home/whoiam/projects/whook/dist/services/DEV_ACCESS_TOKEN",
    ],
    Array [
      "üö´ - Service \\"uniqueId\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/uniqueId",
    ],
    Array [
      "üö´ - Service \\"uniqueId\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/uniqueId",
    ],
    Array [
      "üö´ - Service \\"uniqueId\\" not found in: /home/whoiam/projects/whook/dist/services/uniqueId",
    ],
    Array [
      "üíø - Loading API initializer from /home/whoiam/projects/whook/packages/whook-example/src/services/API.ts.",
    ],
    Array [
      "üö´ - Service \\"BASE_URL\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/BASE_URL",
    ],
    Array [
      "üö´ - Service \\"BASE_URL\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/BASE_URL",
    ],
    Array [
      "üíø - Service \\"BASE_URL\\" found in: /home/whoiam/projects/whook/dist/services/BASE_URL.js",
    ],
    Array [
      "üíø - Loading BASE_URL initializer from /home/whoiam/projects/whook/dist/services/BASE_URL.js.",
    ],
    Array [
      "üìá - Process service initialized.",
    ],
    Array [
      "üí± - HTTP Transaction initialized.",
    ],
    Array [
      "üö´ - Service \\"PROTOCOL\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/PROTOCOL",
    ],
    Array [
      "üö´ - Service \\"PROTOCOL\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/PROTOCOL",
    ],
    Array [
      "üö´ - Service \\"PROTOCOL\\" not found in: /home/whoiam/projects/whook/dist/services/PROTOCOL",
    ],
    Array [
      "üö´ - Could not load \\".env.test\\" file.",
    ],
    Array [
      "Error: ENOENT: no such file or directory, open '/home/whoiam/projects/whook/packages/whook-example/.env.test'",
    ],
    Array [
      "üàÅ - Generated the BASE_URL constant:",
      "http://localhost:9999",
    ],
    Array [
      "ü¶Ñ - Initializing the API service!",
    ],
    Array [
      "üíø - Service \\"getOpenAPIWrapped\\" found in: /home/whoiam/projects/whook/packages/whook-example/src/handlers/getOpenAPI.ts",
    ],
    Array [
      "üíø - Service \\"optionsWithCORSWrapped\\" found in: /home/whoiam/projects/whook/packages/whook-example/src/handlers/optionsWithCORS.ts",
    ],
    Array [
      "üö´ - Service \\"getPingWrapped\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/handlers/getPing",
    ],
    Array [
      "üö´ - Service \\"getPingWrapped\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/handlers/getPing",
    ],
    Array [
      "üíø - Service \\"getPingWrapped\\" found in: /home/whoiam/projects/whook/dist/handlers/getPing.js",
    ],
    Array [
      "üíø - Service \\"getDelayWrapped\\" found in: /home/whoiam/projects/whook/packages/whook-example/src/handlers/getDelay.ts",
    ],
    Array [
      "üíø - Service \\"getDiagnosticWrapped\\" found in: /home/whoiam/projects/whook/packages/whook-example/src/handlers/getDiagnostic.ts",
    ],
    Array [
      "üíø - Service \\"getTimeWrapped\\" found in: /home/whoiam/projects/whook/packages/whook-example/src/handlers/getTime.ts",
    ],
    Array [
      "üíø - Service \\"putEchoWrapped\\" found in: /home/whoiam/projects/whook/packages/whook-example/src/handlers/putEcho.ts",
    ],
    Array [
      "üíø - Loading getOpenAPIWrapped initializer from /home/whoiam/projects/whook/packages/whook-example/src/handlers/getOpenAPI.ts.",
    ],
    Array [
      "üíø - Loading optionsWithCORSWrapped initializer from /home/whoiam/projects/whook/packages/whook-example/src/handlers/optionsWithCORS.ts.",
    ],
    Array [
      "üíø - Loading getPingWrapped initializer from /home/whoiam/projects/whook/dist/handlers/getPing.js.",
    ],
    Array [
      "üíø - Loading getDelayWrapped initializer from /home/whoiam/projects/whook/packages/whook-example/src/handlers/getDelay.ts.",
    ],
    Array [
      "üíø - Loading getDiagnosticWrapped initializer from /home/whoiam/projects/whook/packages/whook-example/src/handlers/getDiagnostic.ts.",
    ],
    Array [
      "üíø - Loading getTimeWrapped initializer from /home/whoiam/projects/whook/packages/whook-example/src/handlers/getTime.ts.",
    ],
    Array [
      "üíø - Loading putEchoWrapped initializer from /home/whoiam/projects/whook/packages/whook-example/src/handlers/putEcho.ts.",
    ],
    Array [
      "üíø - Service \\"WRAPPERS\\" found in: /home/whoiam/projects/whook/packages/whook-example/src/services/WRAPPERS.ts",
    ],
    Array [
      "üíø - Loading WRAPPERS initializer from /home/whoiam/projects/whook/packages/whook-example/src/services/WRAPPERS.ts.",
    ],
    Array [
      "üíø - Service \\"MECHANISMS\\" found in: /home/whoiam/projects/whook/packages/whook-example/src/services/MECHANISMS.ts",
    ],
    Array [
      "üö´ - Service \\"DEFAULT_MECHANISM\\" not found in: /home/whoiam/projects/whook/packages/whook-example/src/services/DEFAULT_MECHANISM",
    ],
    Array [
      "üö´ - Service \\"DEFAULT_MECHANISM\\" not found in: /home/whoiam/projects/whook/packages/whook-cli/dist/services/DEFAULT_MECHANISM",
    ],
    Array [
      "üö´ - Service \\"DEFAULT_MECHANISM\\" not found in: /home/whoiam/projects/whook/dist/services/DEFAULT_MECHANISM",
    ],
    Array [
      "üíø - Service \\"authentication\\" found in: /home/whoiam/projects/whook/packages/whook-example/src/services/authentication.ts",
    ],
    Array [
      "üíø - Loading MECHANISMS initializer from /home/whoiam/projects/whook/packages/whook-example/src/services/MECHANISMS.ts.",
    ],
    Array [
      "üíø - Loading authentication initializer from /home/whoiam/projects/whook/packages/whook-example/src/services/authentication.ts.",
    ],
    Array [
      "üîß - Initializing auth mechanisms",
    ],
    Array [
      "üîê - Initializing the authorization wrapper for \\"getOpenAPI\\".",
    ],
    Array [
      "üîê - Initializing the authorization wrapper for \\"optionsWithCORS\\".",
    ],
    Array [
      "üîê - Initializing the authorization wrapper for \\"getPing\\".",
    ],
    Array [
      "üîê - Initializing the authorization wrapper for \\"getDelay\\".",
    ],
    Array [
      "üîê - Initializing the authorization wrapper for \\"getDiagnostic\\".",
    ],
    Array [
      "üîê - Initializing the authorization wrapper for \\"getTime\\".",
    ],
    Array [
      "üîê - Initializing the authorization wrapper for \\"putEcho\\".",
    ],
    Array [
      "üö¶ - HTTP Router initialized.",
    ],
  ],
  "logErrorCalls": Array [
    Array [
      "üñ• - Using an isolated env.",
    ],
    Array [
      "‚ö° - Loading configurations from /home/whoiam/projects/whook/packages/whook-example/src/config/test/config.",
    ],
    Array [
      "üîÇ - Running in \\"test\\" environment.",
    ],
    Array [
      "‚ö†Ô∏è - Using fake auth mechanism!",
    ],
    Array [
      "üíÅ - Serving the public API docs: http://localhost:9999/docs?url=http%3A%2F%2Flocalhost%3A9999%2Fv4%2FopenAPI",
    ],
    Array [
      "üéôÔ∏è - HTTP Server listening at \\"http://localhost:9999\\".",
    ],
    Array [
      "On air üöÄüåï",
    ],
  ],
  "logInfoCalls": Array [],
}
`;
